cranelift/codegen/src/isa/aarch64/inst/args.rs

```diff
@@ -137,11 +137,20 @@ pub enum MemArg {
     /// Scaled (by size of a type) unsigned 12-bit immediate offset from reg.
     UnsignedOffset(Reg, UImm12Scaled),
 
-    /// Offset from the stack pointer. Lowered into a real amode at emission.
+    /// ---- virtual addressing modes that are lowered at emission time ----
+
+    /// Reference to a "label": e.g., a symbol.
+    Label(MemLabel),
+
+    /// Offset from the stack pointer.
     SPOffset(i64),
 
-    /// Offset from the frame pointer. Lowered into a real amode at emission.
+    /// Offset from the frame pointer.
     FPOffset(i64),
+
+    /// Offset from the "nominal stack pointer": the real SP, except when
+    /// it is temporarily adjusted downward to prepare for a call.
```

Comment:
Since this comment will be largely read by everyone who runs into this kind of `MemArg`, it should be more precise. Can you explain what's above and below (closer to FP and further away from FP, to avoid the possible double interpretation of above/below)? Maybe draw some ASCII art to show a stack example and draw where the nominal SP is?

Reply:
Done! Added a diagram in the main module comment for `abi.rs`; hopefully things are clearer now.